dotnet/src/Functions/Functions.UnitTests/Markdown/Functions/KernelFunctionMarkdownTests.cs

```diff
@@ -18,11 +19,63 @@ public void ItShouldCreatePromptFunctionConfigFromMarkdown()
         Assert.NotNull(model);
         Assert.Equal("TellMeAbout", model.Name);
         Assert.Equal("Hello AI, tell me about {{$input}}", model.Template);
-        Assert.Equal(2, model.ExecutionSettings.Count);
+        Assert.Equal(3, model.ExecutionSettings.Count);
         Assert.Equal("gpt4", model.ExecutionSettings["service1"].ModelId);
         Assert.Equal("gpt3.5", model.ExecutionSettings["service2"].ModelId);
     }
 
+    [Fact]
+    public void ItShouldInitializeFunctionChoiceBehaviorsFromMarkdown()
+    {
+        // Arrange
+        var kernel = new Kernel();
+        kernel.Plugins.AddFromFunctions("p1", [KernelFunctionFactory.CreateFromMethod(() => { }, "f1")]);
+        kernel.Plugins.AddFromFunctions("p2", [KernelFunctionFactory.CreateFromMethod(() => { }, "f2")]);
+        kernel.Plugins.AddFromFunctions("p3", [KernelFunctionFactory.CreateFromMethod(() => { }, "f3")]);
+
+        // Act
+        var function = KernelFunctionMarkdown.CreateFromPromptMarkdown(Markdown, "TellMeAbout");
+
+        // Assert
+        Assert.NotNull(function);
+        Assert.NotEmpty(function.ExecutionSettings);
+
+        Assert.Equal(3, function.ExecutionSettings.Count);
+
+        // AutoFunctionCallChoice for service1
+        var service1ExecutionSettings = function.ExecutionSettings["service1"];
+        Assert.NotNull(service1ExecutionSettings?.FunctionChoiceBehavior);
+
+        var autoConfig = service1ExecutionSettings.FunctionChoiceBehavior.GetConfiguration(new FunctionChoiceBehaviorConfigurationContext([]) { Kernel = kernel });
```

Comment:
The kernel here is the one supplied via `IChatCompletionService.GetChatMessageContentsAsync` and `IChatCompletionService.GetStreamingChatMessageContentsAsync` methods, which declare the kernel parameter as an optional one. Therefore, the kernel should be optional here as well.

The empty list passed to the class constructor is actually chat history, not a list of functions. Chat history is required for the methods mentioned above, so it's expected as a parameter in the class constructor. Otherwise, the compiler gives an error: "CS8618: Non-nullable property 'ChatHistory' must contain a non-null value when exiting constructor..."

> I personally prefer having the functions as a init property.
That's the way I initially declared the property, trying to avoid having the constructor. Later, however, because of the warning, I added the constructor with the chat history parameter.

Reply:
Fixed